### ISSUE TITLE
Feature request: deprecate AccountInfo, have people explicitly use UncheckedAccount

### ISSUE THREAD

**Issue by mikemaccana** [coral-xyz/anchor#2794]
Per [the anchor docs](https://docs.rs/anchor-lang/latest/anchor_lang/accounts/index.html):

> AccountInfo can be used as a type but [Unchecked Account](https://docs.rs/anchor-lang/latest/anchor_lang/accounts/unchecked_account/struct.UncheckedAccount.html) should be used instead.

If my understanding is correct,

 - UncheckedAccount is unsafe, and is named appropriately for something that does not perform checks.

 - AccountInfo is unsafe, and has a name that doesn't indicate it should be avoided unless you know what you're doing (I'm aware the name comes from Solana itself not Anchor)

Ideally Anchor should have one name that's explicit for unchecked accounts. AccountInfo should throw up a warning or just be removed in the next breaking release.

This would also mean we don't have to bother explaining this to everyone learning Anchor!

Thanks for listening! 😊

**Comment by acheroncrypto** [comment]
Yes, your understanding is correct. Also, using `UncheckedAccount` is more performant than using `AccountInfo`. This is a reasonable request and in fact, I was already considering deprecating `AccountInfo` before the next release.

**Comment by mikemaccana** [comment]
Thanks Acheron!

**Comment by vladdem0504** [comment]
Cool, but how would you deal with remaining_accounts?

**Comment by johnsaigle** [comment]
@acheroncrypto could you explain how UncheckedAccount is more performant than AccountInfo?
